octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -240,14 +401,29 @@ async def get_symbol_prices(self,
                                 **kwargs: dict) -> typing.Optional[list]:
         try:
             with self.error_describer():
-                if limit:
+                if limit := self.cut_candle_limit(limit):
                     return await self.client.fetch_ohlcv(symbol, time_frame.value, limit=limit,
                                                          since=since, params=kwargs)
                 return await self.client.fetch_ohlcv(symbol, time_frame.value, since=since, params=kwargs)
         except ccxt.NotSupported:
-            raise octobot_trading.errors.NotSupported
+            if limit:
+                raise octobot_trading.errors.NotSupported
         except ccxt.BaseError as e:
-            raise octobot_trading.errors.FailedRequest(f"Failed to get_symbol_prices: {e.__class__.__name__} on {e}")
+            if limit:
+                raise octobot_trading.errors.FailedRequest(
+                    f"Failed to get_symbol_prices: {e.__class__.__name__} on {e}")
+        except Exception as e:
+            if limit:
+                raise octobot_trading.errors.FailedRequest(
+                    f"Failed to get_symbol_prices: Unknown error on {symbol}/{time_frame} {e}")
+        # try again with a limit
+        return await self.get_symbol_prices(symbol=symbol, time_frame=time_frame, since=since,
```

Comment:
With this code, on a exchange that needs to have a limit we will make two requests each time we call this method. Because the first one will always fail (without limit) and the second one will succeed. I don't think it's a good idea to try to handle this call in a generic way

Reply:
However is self.connector_config.CANDLE_LOADING_LIMIT is set, then we don't do 2 requests

Reply:
Yeah should be safe to use with and without a limit